docs/src/main/asciidoc/security.adoc

```diff
@@ -96,12 +98,79 @@ Additionally it provides `JWT Generation API` for creating `signed`, `inner-sign
 
 See the link:security-jwt[Using SmallRye JWT] guide for more information.
 
-=== OAuth2
+==== OAuth2
 
 `quarkus-elytron-security-oauth2` provides an alternative to `quarkus-oidc` Bearer Token Authentication Mechanism. It is based on `Elytron` and is primarily meant for introspecting the opaque tokens remotely.
 
 See the link:security-oauth2[Using OAuth2] guide for more information.
 
+==== Choosing between OpenId Connect, SmallRye JWT and OAuth2 extensions
+
+`quarkus-oidc` extension requires an OpenId Connect provider such as Keycloak which can be used to verify the Bearer tokens or authenticate the human users with the Authorization Code flow. In both cases `quarkus-oidc` requires a connection to this OpenId Connect provider.
```

Comment:
I think 'end user' is  more generic term than human, as it can still be a program emulating a end user (or a well trained monkey)